### DISCUSSION TITLE
rethrowing the error in handleReturnedServerError loses the orignal error

### DISCUSSION THREAD

**Discussion by reymondgomera** [next-safe-action/next-safe-action#244] [Q&A]
I was throwing an error in my backend and then rethrowing it in the handleReturnedServerError function, but for some reason, it loses the original error, such as when I throw a Prisma error.

**Comment by TheEdoRan** [comment]
Can you provide a link to a minimal reproduction of this issue? Thanks.

**Comment by reymondgomera** [comment]
I can't provide the link right now, but what I have is a server action. When the server action encounters an error, like a PrismaClientError, I just throw it, and handleReturnedServerError catches it. Inside handleReturnedServerError, you can still access the original error, but when you rethrow it, the client can no longer access the original error—only the name, digest, and message are accessible.

![image](https://github.com/user-attachments/assets/5edb3072-4d57-4e86-a392-ccf888d9389b)

![image](https://github.com/user-attachments/assets/cc552207-da4d-4059-81dd-637892127749)

![image](https://github.com/user-attachments/assets/70fd071b-816b-4132-b29c-a769fc10e0c7)

**Comment by TheEdoRan** [comment]
I've just tested this by setting up a new project with Prisma + SQLite, and the same thing happens with Server Actions that are not defined using next-safe-action, so I'd say that's not a lib related issue.

**Reply by reymondgomera** [reply]
Hmm.. the server actions that  I am using are defined using next-safe-action, though. Were you able to access the original error on the client side? Even when you are just re throwing the error in the handleReturnedServerError() funtion?

**Reply by TheEdoRan** [reply]
I've tried both ways, an action defined using next-safe-action and a different one which is just an async function. The behavior is the same: `code` prop gets printed on the server, but is undefined on the client.
